### ISSUE TITLE
ScopedBuilder invalid behaviour

### ISSUE THREAD

**Issue by unipenlabs** [Flutterando/triple_pattern#41]
Hello,

thanks for the great package!

There is an error related to the loading / error behaviour on "ScopedBuilder".
Here is the example code:

```
import 'package:flutter/material.dart';
import 'package:flutter_triple/flutter_triple.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(title: 'Tripple Test', home: MyHomePage());
  }
}

class Counter extends StreamStore<Exception, int> {
  Counter() : super(0);

  Future<void> increment() async {
    setLoading(true);

    await Future.delayed(Duration(seconds: 1));

    int value = state + 1;
    if (value < 5) {
      update(value);
    } else {
      setError(Exception('State not can be > 4'));
    }
    setLoading(false);
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key}) : super(key: key);
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final _counter = Counter();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Test'),
      ),
      body: ScopedBuilder<Counter, Exception, int>(
        store: _counter,
        onState: (state, counter) {
          print('onState called: $counter');
          return Padding(
            padding: EdgeInsets.all(10),
            child: Text('$counter'),
          );
        },
        onError: (context, error) {
          print('onError called: $error');
          return Center(
            child: Text(
              error.toString(),
              style: TextStyle(color: Colors.red),
            ),
          );
        },
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: (() async {
          await _counter.increment();
        }),
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}
```

This is the output:

```
Restarted application in 155ms.
onState called: 0
onState called: 0
onError called: null
onState called: 2
onState called: 3
onState called: 4
onState called: 4
onState called: 4
...
```

It never calls onError. If I remove setLoading(true); / setLoading(false); then everything works as expected.

This is my Flutter configuration:

```
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-x64, locale en-GB)
```

Thanks!

**Comment by jacobaraujo7** [comment]
Thanks for comments !
Could you tell me what the version?

**Comment by unipenlabs** [comment]
Hi,

I am sorry. Here is the version in my pubspec.yml:

flutter_triple: ^1.2.0

**Comment by jacobaraujo7** [comment]
Fixed version 1.2.1. Thank you @unipenlabs

**Comment by unipenlabs** [comment]
Great work! Thank you!
